### ISSUE TITLE
Writes corrupted packet into pcap

### ISSUE THREAD

**Issue by krisutofu** [PretendoNetwork/HokakuCafe#5]
I dumped like 20 MiBs of network data. When I opened the ` .pcap`  in the Wireshark GUI, it told me, it would be corrupted because (at least) one packet size (giving a huge number) exceeds the allowed maximum packet size limit (of less than 2¹⁵).

I found out that after 1 ¼ hours of packet capture, one bigger IP packet was corrupted in the pcap. ~~The total packet size is 1414 (total IP datagram length = 1400) bytes but it actually stores all data together as one packet, an actual packet size of 19701 (total IP datagram length = 19687). The IP datagram total length is too small and ends within JPEG raw data.~~

~~Unsupported configuration?~~

~~I noticed, I increased the MaxPacketSize to 0x10000 from 0x1000 previously (I was afraid packets could be truncated). I was unaware that packet sizes are typically never larger than 16-bit size values.~~

``` 
[General]
Enabled=True ; Disables or enables HokakuCafe, can either be True or False

[Net]
Mode=ALL ; Can be either ALL, IPV4, TCP, UDP or PRUDP
MaxPacketSize=0x10000 ; The maximum size written to the file, larger packets will be truncated
```

**Comment by GaryOderNichts** [comment]
Hmm interesting, haven't experienced that before.
Are you able to upload this capture somewhere and DM (in case it contains private information) it to me on Discord (`@garyodernichts`)?

**Comment by krisutofu** [comment]
I took more time to invest the problem in a hex editor.

Well, the problem is much stranger than I thought. I don't know, if it is related to Hokakucafe. Actually, JPEG image raw data should not appear in the `.pcap` file. I would expect packet data only (which is encrypted). I noticed, 8 screenshot images that I took from the gamepad had overridden data from the `.pcap` file. The images start at very aligned positions, at 0x1000 byte boundaries. Looks like a memory aliasing error or buffer overflow issue.

I remember having made screenshots too quickly in succession so the screenshot plugin showed a red error text (it would still process another screenshot).

If you really want to have these 20 MiB of pcap, I can send them to you.

**Comment by GaryOderNichts** [comment]
That's odd, did the screenshots save successfully on the SD Card or do you also have missing/corrupted screenshots?
I wonder if this is some issue in the FAT driver.

**Comment by krisutofu** [comment]
Indeed, I received over 200 gamepad screenshots successfully on my SD card.

PS: I gonna delete all the packet data that has been overridden by the image file raw data. And hope that I can submit it to Pretendo.

**Comment by jonbarrow** [comment]
Is this consistent? I have used the screenshot plugin with Hokaku at the same time several times. If there's steps to replicate that would be helpful

**Comment by krisutofu** [comment]
From 3 times (3 files) capturing Super Mario Maker "sessions", only the largest file has this issue.

Unfortunately, I misnamed the title. It seems to be a memory problem when making screenshots. I found 8 screenshot images, whose timestamps match the overwritten packet times with a tolerance of single seconds, inside the > 20MiB of `.pcap` (from over 200 screenshots during that capture).

It first appeared to be random, not sure, if it is a race condition for SD card memory blocks. I did not find this issue when I used MaxPacketSize 4KiB but those captures were much smaller. The first corruption happened after 8 ¾ MiB in the file.

I should keep my playing to fewer hours and it should be fine.

**Comment by krisutofu** [comment]
> Is this consistent? I have used the screenshot plugin with Hokaku at the same time several times. If there's steps to replicate that would be helpful

Sorry, I just saw, I did not answer you. Yep, I had this problem multiple times. It happened again last Sunday where we tried to squeeze out multiple games. Maybe it happens randomly? Maybe it does not happen when using the Tiramisu version (which I did not test)? Maybe it's rather a bug in the screenshots plugin because I did not find HokakuCafe to corrupt any outside files yet.
